src-java/base-topology/base-storm-topology/src/test/java/org/openkilda/wfm/share/mappers/HistoryMapperTest.java

```diff
@@ -0,0 +1,157 @@
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.share.mappers;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.openkilda.messaging.payload.history.FlowDumpPayload;
+import org.openkilda.messaging.payload.history.FlowHistoryEntry;
+import org.openkilda.messaging.payload.history.FlowHistoryPayload;
+import org.openkilda.model.FlowEncapsulationType;
+import org.openkilda.model.PathComputationStrategy;
+import org.openkilda.model.SwitchId;
+import org.openkilda.model.history.FlowEvent;
+import org.openkilda.model.history.FlowEventAction;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mapstruct.factory.Mappers;
+
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+
+public class HistoryMapperTest {
+    public static final HistoryMapper INSTANCE = Mappers.getMapper(HistoryMapper.class);
+
+    public static Instant TIMESTAMP = Instant.now();
+    public static String ACTION = "Action";
+    public static String TASK_ID = "Task ID";
+    public static String DETAILS = "Details";
+    public static String ACTOR = "Actor";
+    public static String AFFINITY_GROUP_ID = "1";
+    public static boolean ALLOCATE_PROTECTED_PATH = true;
+    public static int BANDWIDTH = 2;
+    public static int DESTINATION_INNER_VLAN = 3;
+    public static int DESTINATION_PORT = 4;
+    public static String DESTINATION_SWITCH = "5";
+    public static int DESTINATION_VLAN = 6;
+    public static String DIVERSE_GROUP_ID = "7";
+    public static FlowEncapsulationType FLOW_ENCAPSULATION_TYPE = FlowEncapsulationType.VXLAN;
+    public static int FORWARD_COOKIE = 8;
+    public static Long FORWARD_METER_ID = 9L;
+    public static String FORWARD_PATH = "10";
+    public static String FORWARD_STATUS = "11";
+    public static boolean IGNORE_BANDWIDTH = false;
+    public static SwitchId LOOP_SWITCH_ID = new SwitchId(12);
+    public static int MAX_LATENCY = 13;
+    public static PathComputationStrategy PATH_COMPUTATION_STRATEGY = PathComputationStrategy.COST;
+    public static boolean PERIODIC_PINGS = false;
+    public static boolean PINNED = false;
+    public static int REVERSE_COOKIE = 14;
+    public static long REVERSE_METER_ID = 15;
+    public static String REVERSE_PATH = "16";
+    public static String REVERSE_STATUS = "17";
+    public static int SOURCE_INNER_VLAN = 18;
+    public static int SOURCE_PORT = 19;
+    public static String SOURCE_SWITCH = "20";
+    public static int SOURCE_VLAN = 21;
+    public static String TYPE = "22";
+
+    public static FlowEventAction action = new FlowEventAction();
```

Comment:
i think it's better to put `= new FlowEventAction();` into `initializeData`
As you did with `event`

Reply:
Agree with you, changed